template/FormlessTemplate.purs

```diff
@@ -0,0 +1,129 @@
+module FormlessTemplate where
+
+import Prelude
+
+import DOM.HTML.Indexed.InputType (InputType(..))
+import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Effect.Aff (Aff)
+import Formless as F
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Template.DataAndValidation (NAME_FIELD, _name, minLength)
+import Type.Row (type (+))
+
+type FormFieldsRow f =
+  ( NAME_FIELD f
+  + ()
+  )
+
+type FormFields = { | FormFieldsRow F.OutputType }
+
+newtype Form r f = Form (r (FormFieldsRow f))
+derive instance newtypeForm' :: Newtype (Form r f) _
+
+-- Form component types
+
+type Input = Unit
+type AddedState = ( additionalState :: Maybe Int )
+data Action
+  = DoStuff
+  | Initialize
+  | Finalize
+  | Receive Input
+
+data Query a
+  = Command a
+  | Reply (Unit -> a)
+
+type Message = FormFields
+type ChildSlots =
+  ()
+type MonadType = Aff
```

Comment:
I think I'd rather this remain polymorphic, `m`, as that's the most common way the component is used in practice. Then just `component :: forall m. MonadAff m => ... m`